src/drive/web/modules/move/MoveModal.jsx

```diff
@@ -106,22 +109,33 @@ const MoveModal = ({ onClose, entries, classes }) => {
   const handleConfirmation = async () => {
     const sharedParentPath = getSharedParentPath(entries[0].path)
     const targetPath = await CozyFile.getFullpath(folderId, entries[0].name)
-    if (sharedParentPath !== null) {
-      if (targetPath.startsWith(sharedParentPath)) {
-        moveEntries()
-      } else {
-        setMovingOutsideSharedFolder(true)
-      }
-    } else {
-      if (
-        hasOneOfEntriesShared(entries, byDocId) &&
-        hasSharedParent(targetPath)
-      ) {
-        setMovingSharedFolderInsideAnother(true)
-      } else {
-        moveEntries()
-      }
+
+    const areMovedFilesShared = hasOneOfEntriesShared(entries, byDocId)
+    const isOriginParentShared = hasSharedParent(entries[0].path)
+    const isTargetShared = hasSharedParent(targetPath)
+    const isInsideSameSharedFolder = targetPath.startsWith(sharedParentPath)
+
+    if (isInsideSameSharedFolder) {
+      moveEntries()
```

Comment:
As this is the same effect as the default path (line 138), does this means that this if/return condition is necessary to be called before other ifs?

Reply:
Yes this condition should be called before every others. If we are moving a file into the same shared folder we don't need to warn the user. @Ldoppea you see this list of conditions differently ?

Reply:
No, I just wanted to be sure to understand this correctly